tests/test_signed.rs

```diff
@@ -0,0 +1,265 @@
+#![feature(plugin, custom_attribute)]
+#![plugin(rustproof)]
+#![allow(dead_code)]
+#![allow(unused_attributes)]
+fn main() { }
+
+// FIXME: some preconditions are unnecisarilly restricitve ie add_five_i32_invalid does not need x: i32 >= i32::MIN + 5i32
+
+// * * *
+// Integer Add Tests
+// * * *
+
+// Tests signed integer overflow for 32 bit integers
+// Should be invalid
+#[condition(pre="(x: i32 > 0) && (x: i32 < 10)", post="return: (x: i32 > 30)")]
```

Comment:
the postcondition `return: (x: i32 > 30)` doesn't make sense to me.
the format is `<variable>:<type> <operator> <value>`
to me it seems as if you are describing returns type as `(x: i32 > 30)`
I am surprised this doesn't cause a parsing error.
@salterm should take a look at this. I suspect it is because our parser is more permissive than it needs to be. I think this might have been a design decision at some point for some reason.

Also, this test doesn't test for signed integer overflow like the comment implies. In your precondition you restrict x to greater than 0 && less than 10; any value of [1..9] plus or minus five can fit in an i32.

For example to test for overflow I would be inclined to write:
```
// Tests signed integer overflow catching for 32 bit integers
// Should be invalid
#[condition(pre="x:i32 <= i32::MAX - 4i32", post="return: i32 == (x:i32 + 5i32)")]
fn add_five_i32_invalid(x: i32) -> i32 {
    x+5
}
```